src/lib.rs

```diff
@@ -351,3 +460,59 @@ fn convert_response_to_query_response(res: skar_client::QueryResponse) -> Result
         },
     })
 }
+
+/// Construct response and centralize error mapping for calling function.
+fn compose_pyarrow_response(
+    archive_height: Option<u64>,
+    next_block: u64,
+    total_execution_time: u64,
+    blocks: PyObject,
+    transactions: PyObject,
+    logs: PyObject,
+) -> Result<QueryResponseArrow> {
+    Ok(QueryResponseArrow {
+        archive_height: archive_height
+            .map(|h| h.try_into())
+            .transpose()
+            .context("convert height")?,
+        next_block: next_block.try_into().context("convert next_block")?,
+        total_execution_time: total_execution_time
+            .try_into()
+            .context("convert total_execution_time")?,
+        data: QueryArrowResponse {
+            blocks,
+            transactions,
+            logs,
+        },
+    })
+}
+
+/// Uses RecordBatchReader to convert vector of ArrayBatch to reader by c-interface
+/// and then crates table from this reader with method from_batches
+fn convert_batch_to_pyarrow_table<'py>(
+    py: Python<'py>,
+    pyarrow: &'py PyModule,
+    batches: Vec<ArrowBatch>,
+) -> PyResult<PyObject> {
+    let schema = batches.first().unwrap().schema.fields.clone();
+    let field = Field::new("a", DataType::Struct(schema), true);
+
+    let mut data = vec![];
+    for batch in batches {
+        data.push(
+            StructArray::new(field.data_type.clone(), batch.chunk.arrays().to_vec(), None).boxed(),
+        );
+    }
+
+    let iter = Box::new(data.into_iter().map(Ok)) as _;
+    let stream = Box::new(ffi::export_iterator(iter, field));
+    let py_stream = pyarrow.getattr("RecordBatchReader")?.call_method1(
+        "_import_from_c",
+        ((&*stream as *const ffi::ArrowArrayStream) as Py_uintptr_t,),
+    )?;
+    let table = pyarrow
```

Comment:
This is creating a table with a single column which is a struct type if I understand correctly.

What we probably want to do is move the List(RecordBatchReader) to python land and convert it into a table with proper columns in `__init__.py`

We can skip this in this PR as we should overhaul how all thouse response types are returned into python.

Reply:
This is the result value:
`res.data.blocks
pyarrow.Table
number: uint64 not null
hash: binary not null
timestamp: binary not null
----
number: [[17123123,17123124,17123125,17123126,17123127,...,17123218,17123219,17123220,17123221,17123222]]
hash: [[8F318F7FA7CE9DA6CC3A85A20621A7AA11204DE983179BD14AFFE238B25A8AA3,233F0779D81AE0A1AB2C1720E84F5F140D81FB3763E8F6D6B1AFEF08E40E0275,3DC04D23E3FE97B549EEB58FDBA86C25D93F4C6D65CF925C81BDF8C5AFE733F6,A9EEA93A584DA6C9ECB2E17707CF5B7CE1E75FFCE72B09F130447AC0C98B4000,00B4A30A46F6AF259E37F31E26FF40BF82D531FFA7BC28A551BD64EB4BCD9364,...,5389ECD4E6D435C9824BA8BA2FE65E0CB654569C4DE17999EAA502073F6E0D60,C166E15CB207963E67753C293E8982C24FFE2CDE6A757D8849A353C5FF59F711,7D500E10996CB093182B72FDD24888BAE013FCFD3E2E4EB534B828265A07C2D5,A2C490F458ADE488C95B4D176427644A31ED67520C256AE0430E7101288BF03E,21F26B3D1A45464CB54E51527B6B63D61605365E5EF03A0823AF2E4B050BB723]]
timestamp: [[6447C047,6447C053,6447C05F,6447C06B,6447C077,...,6447C4DF,6447C4EB,6447C4F7,6447C503,6447C50F]]
`

Reply:
nice, that looks right